design/ceph-external-cluster.md

```diff
@@ -0,0 +1,115 @@
+
+The flex driver would also be agnostic of the cluster for the same reasons, but we won’t need to worry about the flex driver going forward.
+
+## Rook-Ceph Cluster CRD
+
+In order for Rook to provide the storage to clients in the local cluster, the CephCluster CRD will be created in order for the operator to provide local management of the external cluster. There are several differences needed for the operator to be aware of an external cluster.
+
+1. Before the CephCluster CRD is created, some metadata must be initialized in local configmaps/secrets to allow the local cluster to manage the external cluster.
+   * mon endpoint(s) and admin keyring
+1. The mon, mgr, and osd daemons will not be managed by the local Rook operator. These daemons must be created and managed by the external cluster.
+1. The operator will make a “best effort” to keep the list of mons updated.
+   * If the mons change in the external cluster, the list of mons must be updated in the local cluster.
+   * The operator will need to query the Ceph status periodically (perhaps every minute). If there is a change to the mons, the operator will update the local configmaps/secrets.
+   * If the local operator fails to see changes to the external mons, perhaps because it is down, the mon list could become stale. In that case, the admin will need to update the list similarly to how it was initialized when the local cluster was first created.
+
+The first bullet point above requires an extra manual configuration step by the cluster admin from what he needs in a typical Rook cluster. The other items above will be handled automatically by the Rook operator. The extra step involves exporting metadata from the external cluster and importing it to the local cluster:
+1. The admin creates a yaml file with the needed resources from the external cluster (ideally we would provide a helper script to help automate this task):
+   * Save the mon list and admin keyring
+1. Load the yaml file into the local cluster
+   * `oc create -f <config.yaml>`
+
+The CephCluster CRD will have a new property “external” to indicate whether the cluster is external. If true, the local operator will implement the described behavior.
+Other CRDs such as CephBlockPool, CephFilesystem, and CephObjectStore do not need this property since they all belong to the cluster and will effectively
+inherit the external property.
+
+```yaml
+kind: CephCluster
+spec:
+  external: true
+```
+
+The mgr modules, including the dashboard, would be running in the external cluster. Any configuration that happens through the dashboard would depend on the orchestration modules in that external cluster.
+
+## Block Storage
+
+With the rook-ceph cluster created, the CSI driver integration will cover the Block (RWO) storage and no additional management is needed.
+
+### Pool
+
+When a pool CRD is created in the local cluster, the operator will create the pool in the external cluster. The pool settings will only be applied the first
```

Comment:
I don't think this https://github.com/rook/rook/pull/2677/files#r258730112 has been captured here.

Reply:
Yeah, i'll add more details, thanks